File: backend-python/src/controllers/bus_controller.py
from fastapi import APIRouter, HTTPException
from src.config.database import get_pool

router = APIRouter(prefix="/buses", tags=["buses"])


@router.get("")
async def get_active_buses():
    pool = await get_pool()
    rows = await pool.fetch("SELECT * FROM active_trips_view ORDER BY route_number")
    return {"trips": [dict(r) for r in rows]}


@router.get("/{bus_id}")
async def get_bus_by_id(bus_id: int):
    pool = await get_pool()
    row = await pool.fetchrow("SELECT * FROM buses WHERE id = $1", bus_id)
    if not row:
        raise HTTPException(status_code=404, detail="Bus not found")
    return {"bus": dict(row)}

File: backend/src/controllers/busController.js
const pool = require('../config/database');

// Get all active buses
const getActiveBuses = async (req, res) => {
  try {
    const result = await pool.query(
      'SELECT * FROM active_trips_view ORDER BY route_number'
    );
    res.json({ trips: result.rows });
  } catch (error) {
    console.error('Get active buses error:', error);
    res.status(500).json({ error: 'Failed to fetch buses' });
  }
};

// Get bus details
const getBusById = async (req, res) => {
  try {
    const { id } = req.params;
    const result = await pool.query('SELECT * FROM buses WHERE id = $1', [id]);
    
    if (result.rows.length === 0) {
      return res.status(404).json({ error: 'Bus not found' });
    }
    
    res.json({ bus: result.rows[0] });
  } catch (error) {
    console.error('Get bus error:', error);
    res.status(500).json({ error: 'Failed to fetch bus' });
  }
};

module.exports = { getActiveBuses, getBusById };
